rakelib/login.rake

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+namespace :login do
+  desc 'Login as <some test user>, then run this task to validate that LOA1 credentials are set properly'
+  task idme_loa_1: [:environment] do
```

Comment:
Great start! I have two main thoughts so far.

I'm wondering if it would be a lot more flexible if we let the user pass in the account identifier as a param, that way we could log in and test on whatever account we want? Ideally we'd want to look them up by email but we could start with `uuid` as a simpler & safer alternative.

Also thinking we should break up the tests into blocks based on IDP, then we can run an IDP's suite of tests all together when we're logged in on the right account for it. If it gets big enough we might even think about breaking out each IDP's test into their own file.

Thinking something like this:
```
task :idme, [:uuid] => [:environment] do
  user = User.find(args[:uuid])
  //run your basic 'did it find a user' validation here
  Rake::Task["login:idme_loa_1"].invoke
  Rake::Task["login:idme_loa_3"].invoke

Reply:
I made the change to accept `uuid` as an argument, but I feel like we should hold off on trying to group tests at this point. It's a good idea, and we definitely should group tests in that way at some point, but as it stands now the manual process this task relies on makes running LOA1 and LOA3 tests together as a package kind of impossible. It's difficult to simulate logging in as more than one user at a time and you would expect opposing results in a lot of cases. I think as we write additional and more complex tasks we'll be able to more naturally judge how they should be grouped later.

Is there anything else you think we should be testing for an ID.me LOA1 user? I'm not sure what else would be valuable to check.

Reply:
Right, makes sense to keep it more straightforward right now. As for other things to test for LoA1 I can't think of anything super important except maybe also checking the AuthN context that comes back to us?

```
validate(
      user.authn_context == LOA::IDME_LOA1_VETS,
      'User has an ID.me LoA1_Vets Authn Context',
      'User does not have an ID.me LoA1_Vets Authn Context'
    )
```